src/browser/NativeMessageInstaller.cpp

```diff
@@ -0,0 +1,313 @@
+    case SupportedBrowsers::BRAVE:
+        return TARGET_DIR_BRAVE;
+    case SupportedBrowsers::EDGE:
+        return TARGET_DIR_EDGE;
+    default:
+        return QString();
+    }
+}
+
+/**
+ * Returns the browser name
+ * Needed for Windows to separate Chromium- or Firefox-based scripts
+ *
+ * @param browser Selected browser
+ * @return QString Name of the selected browser
+ */
+QString NativeMessageInstaller::getBrowserName(SupportedBrowsers browser) const
+{
+    switch (browser) {
+    case SupportedBrowsers::CHROME:
+        return "chrome";
+    case SupportedBrowsers::CHROMIUM:
+        return "chromium";
+    case SupportedBrowsers::FIREFOX:
+        return "firefox";
+    case SupportedBrowsers::VIVALDI:
+        return "vivaldi";
+    case SupportedBrowsers::TOR_BROWSER:
+        return "tor-browser";
+    case SupportedBrowsers::BRAVE:
+        return "brave";
+    case SupportedBrowsers::EDGE:
+        return "edge";
+    default:
+        return QString();
+    }
+}
+
+/**
+ * Returns the path of native messaging JSON script for the selected browser
+ *
+ * @param browser Selected browser
+ * @return QString JSON script path for the selected browser
+ */
+QString NativeMessageInstaller::getNativeMessagePath(SupportedBrowsers browser) const
+{
+    QString basePath;
+#if defined(Q_OS_WIN)
+    // If portable settings file exists save the JSON scripts to the application folder
+    if (QFile::exists(QCoreApplication::applicationDirPath() + QStringLiteral("/keepassxc.ini"))) {
+        basePath = QCoreApplication::applicationDirPath();
+    } else {
+        basePath = QStandardPaths::writableLocation(QStandardPaths::DataLocation);
+    }
+    return QString("%1/%2_%3.json").arg(basePath, HOST_NAME, getBrowserName(browser));
+#elif defined(Q_OS_LINUX)
+    if (browser == SupportedBrowsers::TOR_BROWSER) {
+        // Tor Browser launcher stores its config in ~/.local/share/...
+        basePath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
+    } else if (browser == SupportedBrowsers::FIREFOX) {
+        // Firefox stores its config in ~/
+        basePath = QDir::homePath();
+    } else {
+        basePath = QStandardPaths::writableLocation(QStandardPaths::ConfigLocation);
+    }
+#else
+    basePath = QDir::homePath();
+#endif
+    return QString("%1%2/%3.json").arg(basePath, getTargetPath(browser), HOST_NAME);
+}
+
+/**
+ * Gets the path to keepassxc-proxy binary
+ *
+ * @param location Custom proxy path
+ * @return path Path to keepassxc-proxy
+ */
+QString NativeMessageInstaller::getProxyPath() const
+{
+    if (browserSettings()->useCustomProxy()) {
+        return browserSettings()->customProxyLocation();
+    }
+
+    QString path;
+#ifdef KEEPASSXC_DIST_APPIMAGE
+    path = QProcessEnvironment::systemEnvironment().value("APPIMAGE");
+#else
+    path = QCoreApplication::applicationDirPath() + QStringLiteral("/keepassxc-proxy");
+#ifdef Q_OS_WIN
+    path.append(QStringLiteral(".exe"));
```

Comment:
However, I don't think it's needed here. `QString::append()` has a `const char*` overload. 🤣

Reply:
Consistency ftw!